CHANGELOG.md

```diff
@@ -85,8 +85,14 @@ Certain undersupported and underused Apollo Server features have been removed in
 
 #### Modified functionality
 
-- Apollo Server now always fires the `willSendResponse` lifecycle event after firing `didEncounterError`.
+- With one exception, all Apollo Server plugin methods (`requestDidStart`, `didResolveOperation`, etc.) are now `async`.
+  - Previously, some of these methods were synchronous, others were `async`, and some were "sometimes-`async`" by returning a `ValueOrPromise`.
+  - The exception is `willResolveField`, which remains synchronous. This method is called much more often than any other plugin method, and converting it to `async` might affect performance.
+  - In a future release, `willResolveField` might become "sometimes-`async`" by returning a `ValueOrPromise`.
+- Apollo Server now always fires the `willSendResponse` plugin lifecycle event after firing `didEncounterError`.
   - In certain error cases (mostly related to automated persisted queries), Apollo Server 2 skips firing `willSendResponse`.
+- Renamed the `GraphQLService` interface to `GatewayInterface`.
+  - This interface is the type used to provide a federated gateway instance to Apollo Server. Its name has been changed to reduce ambiguity.
```

Comment:
could note that the old name is still exported for backwards compatibility (so that we can have one version of `@apollo/gateway` that works with both server versions)